src/Estimators/include/BipedalLocomotion/FloatingBaseEstimators/BaseEstimatorFromFootIMU.h

```diff
@@ -0,0 +1,183 @@
+         */
+        class BaseEstimatorFromFootIMU
+            : public BipedalLocomotion::System::Advanceable<BaseEstimatorFromFootIMUInput, BaseEstimatorFromFootIMUState>
+        {
+        public:
+            // stance foot reference frame
+            enum class StanceFootRF
+            {
+                l_sole,
+                r_sole
+            };
+
+            /*
+             * Struct holding parameters for the BaseEstimatorFromFootIMU
+             */
+            struct Params
+            {
+                //nothing here at the moment
+            };
+
+            /*
+             * Set the iDynTree::Model of the robot to be used by the estimator.
+             * @return true in case of success, false otherwise.
+             */
+            bool setModel(iDynTree::Model& model);
+
+            /*
+             * Set the BaseEstimatorFromFootIMU::StanceFootRF name of the stance foot Reference Frame.
+             * Possible options for ergoCub:
+             * --> l_sole
+             * --> r_sole
+             * @return true in case of success, false otherwise.
+             */
+            bool setStanceFootRF(BaseEstimatorFromFootIMU::StanceFootRF frameName);
+
+            /*
+             * Get the BaseEstimatorFromFootIMU::StanceFootRF name of the stance foot Reference Frame.
+             * Possible options for ergoCub:
+             * --> l_sole
+             * --> r_sole
+             * To be used for debug purposes.
+             */
+            StanceFootRF getStanceFootRF() const         { return m_stanceFootRF; }
+
+            /*
+             * Convert the enum class StanceFootRF into an std::string.
+             * @return std::string containing the frame name.
+             */
+            static std::string STANCE_FOOT_TO_STRING(StanceFootRF f);
+
+            /*
+             * Initialize the BaseEstimatorFromFootIMU block.
+             * The setModel and setStanceFootRF methods must be called before initialization.
+             * @param handler pointer to the parameter handler.
+             * @note The following parameters are required
+             * |   Parameter Name    |     Type      | Mandatory |                       Description                                          |
+             * |:-------------------:|:-------------:|:---------:|:--------------------------------------------------------------------------:|
+             * |  `foot_width_in_m`  | `double`      |    Yes    |   Lateral dimension of the robot foot rectangular plate (1 plate)          |
+             * |  `foot_length_in_m` | `double`      |    Yes    |   Frontal dimension of the robot foot rectangular plates (2 plates + gap)  |
+             * |  `base_frame`       | `std::string` |    Yes    |   Name of the root link Reference Frame whose pose must be estimated       |
+             * @return true in case of success/false otherwise.
+             */
```

Comment:
You can pass the foot frame as a parameter in the list. Moreover, I would avoid mentioning that the lateral and frontal dimension of the foot is a rectangular plate + gap.
Moreover, this assumes that the `l_sole` and `r_sole` are placed in the center of the foot. What if this is not true? You can consider the corners of the foot as is done in 

https://github.com/ami-iit/bipedal-locomotion-framework/blob/f4b74fe1151c50f79d62cdc9c57599352e336f6e/src/ML/include/BipedalLocomotion/ML/MANNAutoregressive.h#L141-L144

Reply:
Ok. Everything done, except managing the case in wich `l_sole` and `r_sole` are not placed in the center of the foot.

Reply:
Resolving since this comment is duplicated.